bin/zopen-importenvs

```diff
@@ -1,18 +1,96 @@
-#!/bin/sh
-#
-# Used for development to source all the boot .envs
-# . importenvs
-
-if [ ! -d "${HOME}/zopen/boot/" ]; then
-  echo "boot directory does not exist.  Existing"
-  exit 4;
+#!/bin/env bash
+# bash is required because BASH_SOURCE is used here to locate the sourced script 
+# Used for development to source all the boot/prod .envs
+
+
+(return 0 2>/dev/null) && sourced=1 
+# What exit code did that give?
+if [[ $sourced -ne 1 ]]
+then
+    echo "This script is not sourced."
+    echo "USAGE: . ./zopen-importenvs <path to buildenv to fetch dependency>"
+    exit 6     
 fi
 
-for deps in $(find "${HOME}/zopen/boot/" -name ".env")
-do
-  echo "Found ${deps}"
-  depdir=$(dirname "${deps}")
-  cd $depdir
-  . ./.env
-  cd -
-done
+# below code fetches the ~/code/utils/bin path where the common.inc file is under its lib dir
+export bindir="$( cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd -P )"
+. "${bindir}/lib/common.inc"
+
+#depsPath has the details of the location of ports whose env files are consumed for sourcing 
+depsPath="$HOME/zopen/prod|$HOME/zopen/boot"
+
+# if the number of arguments passed to script is 0,
+# then , the ports are fetched from $HOME/zopen/prod and $HOME/zopen/boot directory
+# then its env files are fetched and sourced, if they are not already sourced.
+if [ $# -eq 0 ]
+then
+     echo "Sourcing projects from $HOME/zopen/prod and $HOME/zopen/boot"
+     
+     # arrOfDirs, is an array which holds the names of dirs whose env is set
+     declare -a arrOfDirs
+     unset arrOfDirs[@]
+
+     #Loop through zopen/boot & zopen/prod
+     for path in $(echo ${depsPath} | sed "s/|/ /g") ; do
+         for deps in $(find "${path}" -name ".env")
+         do
+          depdir=$(dirname "${deps}")
+          dirName="$(basename ${depdir})"
+
+          #If env from this directory is not sourced then, fetch the env file and source it 
+          if [[ ! " ${arrOfDirs[*]} " =~ " ${dirName} " ]]; then
+             # array doesn't contain hence, source the env file
+             echo "Setting up ${depdir} dependency environment"
+             cd $depdir
+             . ./.env
+
+             # If sourcing is successful then, add to array
+             if [ $? -eq 0 ]; then
+                arrOfDirs+=($dirName)
+             fi
+             cd - 
+           else
+             echo "Skipping the set path = $dirName, as it is already sourced" 
+           fi
+         done
+      done 
+else 
+    #If path is provided then builedenv file is fetched from the provided path 
+    pathForBuild=$1
+    if [ ! -d "${HOME}/zopen/boot/" ] && [ ! -d "${HOME}/zopen/prod/" ]; then
+       echo "boot & prod directories do not exist.  Exiting"
```

Comment:
I suggest adding the full paths here

Reply:
Ok, i have added full path in echo output.